packages/neon/neon/lib/src/app.dart

```diff
@@ -122,34 +122,37 @@ class _NeonAppState extends State<NeonApp> with WidgetsBindingObserver, tray.Tra
           if (account == null) {
             return;
           }
-          final app = Provider.of<List<AppImplementation>>(context, listen: false).find('notifications');
-          if (app != null) {
-            await _accountsBloc.getAppsBlocFor(account).getAppBloc<NotificationsBlocInterface>(app).refresh();
-          }
-        };
-        Global.onPushNotificationClicked = (final pushNotificationWithAccountID) async {
+
           final allAppImplementations = Provider.of<List<AppImplementation>>(context, listen: false);
+          final app = allAppImplementations.find('notifications') as NotificationsAppInterface?;
 
-          final app = (pushNotificationWithAccountID.subject.app != null
-                  ? allAppImplementations.find(pushNotificationWithAccountID.subject.app!)
-                  : null) ??
-              allAppImplementations.find('notifications');
+          if (app == null) {
+            return;
+          }
 
+          await _accountsBloc.getAppsBlocFor(account).getAppBloc<NotificationsBlocInterface>(app).refresh();
+        };
+        Global.onPushNotificationClicked = (final pushNotificationWithAccountID) async {
           final account = _accountsBloc.accounts.value.find(pushNotificationWithAccountID.accountID);
           if (account == null) {
             return;
           }
-
           _accountsBloc.setActiveAccount(account);
-          if (app != null) {
-            if (app.id != 'notifications') {
-              _accountsBloc
-                  .getAppsBlocFor(account)
-                  .getAppBloc<NotificationsBlocInterface>(app)
-                  .deleteNotification(pushNotificationWithAccountID.subject.nid!);
-            }
-            await _openAppFromExternal(account, app.id);
+
+          final allAppImplementations = Provider.of<List<AppImplementation>>(context, listen: false);
+          final app = allAppImplementations.find(pushNotificationWithAccountID.subject.app ?? 'notifications');
+
+          if (app == null) {
+            return;
+          }
+
+          if (app is NotificationsAppInterface) {
+            _accountsBloc
+                .getAppsBlocFor(account)
+                .getAppBloc<NotificationsBlocInterface>(app)
+                .deleteNotification(pushNotificationWithAccountID.subject.nid!);
           }
```

Comment:
I''m still not sure about this change but couldn't really debug it either.

Maybe you can pick up these changes.